apps/web/src/pages/UserEdit/UserEdit.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { useRef, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useRouter } from 'next/router';
+import toast from 'react-hot-toast';
+import * as yup from 'yup';
+
+import { Button, FormInput, FormTextArea, LoaderOverlay } from 'ui';
+import AvatarDropzone from 'src/components/AvatarDropzone';
+
+import {
+  useGetCurrentUserQuery,
+  useUpdateUserMutation,
+  useUploadImageMutation,
+} from 'apollo-hooks';
+
+import {
+  MainWrapper,
+  Container,
+  Wrapper,
+  ProfileImageWrapper,
+  ProfileImageButtonWrapper,
+  FormDetails,
+  SubText,
+  SaveProfileWrapper,
+  FlexRowWrapper,
+} from './style';
+import { NextSeo } from 'next-seo';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+const validationSchema = yup
+  .object()
+  .shape({
+    profileName: yup.string().required('This is required field'),
+    profileWebsite: yup.string().url('It must be a valid URL'),
+  })
+  .required();
+
+const UserEdit = () => {
+  let showOverlay = false;
+  const notify = () => toast.success('Profile succesfully updated');
+
+  const dropzoneRef = useRef<any>(null);
+  const router = useRouter();
+  const { data } = useGetCurrentUserQuery();
+  const userDetails = data?.getCurrentUser;
+
+  const {
+    register,
+    watch,
+    getValues,
+    setValue,
+    handleSubmit,
+    reset,
+    formState: { errors, dirtyFields },
+  } = useForm({
+    defaultValues: {
+      preview: userDetails?.avatar,
+      profileName: userDetails?.name,
+      profileBio: userDetails?.bio,
+      profileLocation: userDetails?.location,
+      profileWebsite: userDetails?.website,
+      profileTwitter: userDetails?.twitter,
+      profileDiscord: userDetails?.discord,
+    },
+    resolver: yupResolver(validationSchema),
+  });
+
+  const [updateUser, { loading: updateUserLoading }] = useUpdateUserMutation();
+  const [uploadImage, { loading: imageUploading }] = useUploadImageMutation();
+
+  const onSubmit = async () => {
+    let res, data;
+
+    try {
+      if (JSON.stringify(dirtyFields) == '{}') {
+        return;
+      }
+
+      if (dirtyFields.preview) {
+        const reader = new FileReader();
+        reader.readAsDataURL(getValues('preview'));
+        reader.onload = async () => {
+          res = await uploadImage({
+            variables: {
+              path: String(reader.result),
+            },
+          });
+          if (res) {
+            data = await updateUserVariables(res);
+            await router.push(`/user/${data?.data?.updateUser?.id}`);
+            await notify();
```

Comment:
move above the route change and remove the await from `router.push` and `notify`, they don't return a promise so there is no point in await them..

Reply:
done